src/LSST/control/NetCommandDefs.cpp

```diff
@@ -73,6 +76,91 @@ bool NCmdSwitchCommandSource::action() {
     return result;
 }
 
+
+NCmdPower::Ptr NCmdPower::create(JsonPtr const& inJson_) {
```

Comment:
I'm not sure a shortcut type definition `JsonPtr` is better than `std::shared_ptr<nlohmann::json>`. If the long type name `nlohmann::json` is a problem then one can always do (in the CPP files):
```
namespace json = nlohmann::json;
```
Besides, if you also do:
```
using namespace std;
```
Then the following would look not so long:
```
NCmdPower::Ptr NCmdPower::create(shared_ptr<json> const& inJson_) {
```
Of course, you would have to use the fully-qualified names in the header files:
```
class NCmdPower {
public:
    static Ptr create(std::shared_ptr<nlohmann::json> const& inJson_);
};
```

Reply:
I really do like the convenience of the typedef. `JsonPtr` is pretty clear and, at least me, makes the code more readble.